File: programmers/해쉬/완주하지 못한 선수/javascript/result.js
/**
    정확성: 50.0
    효율성: 0.0
    합계: 50.0 / 100.0
 */
function test1(participant, completion) {
    var answer = '';

    for(var i=0; i<completion.length; i++){
        participant.splice(participant.indexOf(completion[i]),1);
    }

    answer = participant[0];

    return answer;
}

//정답
function solution(participant, completion) {
    participant.sort();
    completion.sort();

    for(var i=0; i<completion.length; i++){
        if(completion[i] != participant[i]){
            return participant[i];
        }
    }

    return participant[participant.length-1];
}

a = solution(["leo", "kiki", "eden"],["eden", "kiki"]); // 결과: leo
b = solution(["marina", "josipa", "nikola", "vinko", "filipa"],["josipa", "filipa", "marina", "nikola"]); // 결과: vinko
c = solution(["mislav", "stanko", "mislav", "ana"],["stanko", "ana", "mislav"]); // 결과: mislav

console.log(a);
console.log(b);
console.log(c);

File: programmers/해쉬/완주하지 못한 선수/java/result.java
package result;

import java.util.*;

public class result {
	public static String solution(String[] participant, String[] completion) {
		String answer = "";
        HashMap<String, Integer> hm = new HashMap<>();
        for (String player : participant) hm.put(player, hm.getOrDefault(player, 0) + 1);
        for (String player : completion) hm.put(player, hm.get(player) - 1);

        System.out.println(hm);
        
        for (String key : hm.keySet()) {
            if (hm.get(key) != 0){
                answer = key;
            }
        }
        return answer;
    }
	
	public static void main(String[] args) {
		String a = solution(new String[] {"leo", "kiki", "eden"},new String[] {"eden", "kiki"});
		String b = solution(new String[] {"marina", "josipa", "nikola", "vinko", "filipa"},new String[] {"josipa", "filipa", "marina", "nikola"});
		String c = solution(new String[] {"mislav", "stanko", "mislav", "ana"},new String[] {"stanko", "ana", "mislav"});
		
		System.out.println(a); // 결과: leo
		System.out.println(b); // 결과: vinko
		System.out.println(c); // 결과: mislav
	}
}
